datastream-kafka-connector/src/test/java/com/linkedin/datastream/connectors/kafka/mirrormaker/TestKafkaConsumerOffsets.java

```diff
@@ -0,0 +1,208 @@
+
+    Map<Integer, Long> partitionOffsets5 = new HashMap<>();
+    partitionOffsets5.put(2, 10L);
+    partitionOffsets5.put(3, 10L);
+    topicPartitionOffsets3.put(topic1, partitionOffsets5);
+
+    responseList2.add(new KafkaConsumerOffsetsResponse(topicPartitionOffsets3, consumerGroup1));
+
+    // instance 2 consumer group 3
+    Map<String, Map<Integer, Long>> topicPartitionOffsets4 = new HashMap<>();
+
+    Map<Integer, Long> partitionOffsets6 = new HashMap<>();
+    partitionOffsets6.put(0, 30L);
+    topicPartitionOffsets4.put(topic2, partitionOffsets6);
+    responseList2.add(new KafkaConsumerOffsetsResponse(topicPartitionOffsets4, consumerGroup3));
+
+    // reducing responses and asserting correctness
+    Map<String, String> responseMap = new HashMap<>();
+    responseMap.put(instance1, JsonUtils.toJson(responseList1));
+    responseMap.put(instance2, JsonUtils.toJson(responseList2));
+
+    String reducedMapJson = KafkaConnectorDiagUtils.reduceConsumerOffsetsResponses(responseMap, LOG);
+    List<KafkaConsumerOffsetsResponse> responseList =
+        JsonUtils.fromJson(reducedMapJson, new TypeReference<List<KafkaConsumerOffsetsResponse>>() { });
+
+    Assert.assertEquals(responseList.size(), 3); // 3 consumer groups
+
+    KafkaConsumerOffsetsResponse cg1Response = responseList.stream().
+        filter(r -> r.getConsumerGroupId().equals(consumerGroup1)).findAny().orElse(null);
+    Assert.assertNotNull(cg1Response);
+    Assert.assertEquals(cg1Response.getConsumerOffsets().keySet().size(), 2); // cg1 consumes both topics
+    Assert.assertEquals(cg1Response.getConsumerOffsets().get(topic1).keySet().size(), 4); // cg1 consumes 4 partitions for topic 1
+    Assert.assertEquals(cg1Response.getConsumerOffsets().get(topic2).keySet().size(), 2); // cg1 consumes 2 partitions for topic 2
+
+    KafkaConsumerOffsetsResponse cg2Response = responseList.stream().
+        filter(r -> r.getConsumerGroupId().equals(consumerGroup2)).findAny().orElse(null);
+    Assert.assertNotNull(cg2Response);
+    Assert.assertEquals(cg2Response.getConsumerOffsets().keySet().size(), 2); // cg2 consumers both topics
+    Assert.assertEquals(cg2Response.getConsumerOffsets().get(topic1).keySet().size(), 2); // cg2 consumes 2 partitions for topic 1
+    Assert.assertEquals(cg2Response.getConsumerOffsets().get(topic2).keySet().size(), 2); // cg2 consumes 2 partitions for topic 2
+
+    KafkaConsumerOffsetsResponse cg3Response = responseList.stream().
+        filter(r -> r.getConsumerGroupId().equals(consumerGroup3)).findAny().orElse(null);
+    Assert.assertNotNull(cg3Response);
+    Assert.assertEquals(cg3Response.getConsumerOffsets().keySet().size(), 1); // cg3 consumes only topic 2
+    Assert.assertEquals(cg3Response.getConsumerOffsets().get(topic2).size(), 1); // cg3 consumes 1 partition for topic 2
+  }
+
+  private boolean testConsumerOffsetsAreUpdated(KafkaMirrorMakerConnector connector) {
```

Comment:
Feel free to ignore this suggestion, but can we take TOPIC_COUNT, PARTITION_COUNT, and PARTITION_MESSAGE_COUNT as input parameters? Reason I ask is that we may want to add more tests in this area in the future, and it'll be nice to make this a future proof utility.

Reply:
For now that can easily be configured by changing the values of constants and rebuilding. If we see the need to add more tests here we can easily parameterize those.